api/filter/http/squash.proto

```diff
@@ -0,0 +1,44 @@
+syntax = "proto3";
+
+// [#proto-status: experimental]
+
+package envoy.api.v2.filter.http;
+
+import "google/protobuf/duration.proto";
+import "validate/validate.proto";
+
+// Squash is an HTTP filter which enables Envoy to integrate with squash microservices debugger.
+// (https://github.com/solo-io/squash). When the squash filter encounters a request containing the
+// header 'x-squash-debug' it will:
+// 1. delay the incoming request.
+// 2. contact the squash server and request the creation of a debug attachment
+//    - On the squash server side, squash will attempt to attach a debugger to the services envoy
+//      proxies to. if it succeeds in doings so, it will changes the state of the debug attachment
+//      to attached.
+// 3. wait until the squash server updates the debug attachment object's state to attached (or
+// error)
+// 4. resume the incoming request
+message SquashConfig {
+  // The name of the cluster that hosts the squash server.
+  string squash_cluster = 1 [(validate.rules).string.min_bytes = 1];
```

Comment:
Nit: I think the `squash_` prefix is redundant, since we're inside the `SquashConfig` message.